internal/abci/abci.go

```diff
@@ -880,6 +898,34 @@ func (a *AbciApp) ProcessProposal(ctx context.Context, req *abciTypes.RequestPro
 }
 
 func (a *AbciApp) Query(ctx context.Context, req *abciTypes.RequestQuery) (*abciTypes.ResponseQuery, error) {
+	if req.Path == statesync.ABCISnapshotQueryPath { // "/snapshot/height"
+		if a.snapshotter == nil {
+			return &abciTypes.ResponseQuery{}, nil
+		}
+
+		var snapshot *snapshots.Snapshot
+		height := string(req.Data)
+		exists := false
+
+		curSnapshots := a.snapshotter.ListSnapshots()
+		for _, s := range curSnapshots {
+			if height == fmt.Sprintf("%d", s.Height) {
+				exists = true
+				snapshot = s
+				break
+			}
+		}
+
+		if !exists {
+			return &abciTypes.ResponseQuery{}, nil
+		}
+
+		bts, err := json.Marshal(snapshot)
```

Comment:
I don't believe this is an issue, but just want to flag: json.Marshal is non-deterministic. I'm not sure if that is an issue here

Reply:
Not a problem. We just need the snapshot metadata to compare, no need it to be deterministic